sktime/annotation/base/_base.py

```diff
@@ -0,0 +1,435 @@
+# -*- coding: utf-8 -*-
+"""
+Base class template for annotator base types:
+ - BaseStreamAnnotator - annotating a data stream, with temporal update
+ - BasePanelAnnotator - annotating a panel of sequences or time series
+
+ Class summaries below:
+
+---
+
+    class name: BasePanelAnnotator
+
+Scitype defining methods:
+    fitting              - fit(self, X, Y=None, Z=None)
+    annotating           - predict(self, X, Z=None)
+
+    NOT YET IMPLEMENTED
+    updating (minibatch) - update(self, X, Y=None, Z=None)
+    update&annotate      - update_predict(self, X, X_new, Y=None, Z=None, Z_new=None)
+
+Inspection methods:
+    hyper-parameter inspection  - get_params()
+    fitted parameter inspection - get_fitted_params()
+
+State:
+    fitted model/strategy   - by convention, any attributes ending in "_"
+    fitted state flag       - check_is_fitted()
+
+---
+
+    class name: BaseStreamAnnotator
+
+Scitype defining methods:
+    fitting              - fit(self, X, Y=None)
+    annotating           - predict(self, X)
+    updating (temporal)  - update(self, X, Y=None)
+    update&annotate      - update_predict(self, X)
+
+Inspection methods:
+    hyper-parameter inspection  - get_params()
+    fitted parameter inspection - get_fitted_params()
+
+State:
+    fitted model/strategy   - by convention, any attributes ending in "_"
+    fitted state flag       - check_is_fitted()
+
+copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""
+
+__author__ = ["satya-pattnaik ", "fkiraly"]
+
+from sktime.base import BaseEstimator
+
+from sktime.utils.validation.series import check_series
+
+import pandas as pd
+
+
+class BasePanelAnnotator(BaseEstimator):
+    """Base annotator
+
+    The base annotator specifies the methods and method
+    signatures that all annotators have to implement.
+
+    Specific implementations of these methods is deferred to concrete
+    annotators.
+    """
+
+    # default tags
+    _tags = {
```

Comment:
Let's add tags in a separate PR once we have agreed on them.

Reply:
ok, I commented them out for now